Update publishing-websites.mdx - added steps for handling dist direct… (#23701)
expo/expo: docs/pages/distribution/publishing-websites.mdx

## Hosting on third-party services

> When deploying from a remote git repository remember to remove the dist directory from the .gitignore file and commit and push the changes. This directory should be selected for the web deployment.

Reviewer: ```suggestion
```

This might not be true for all the deployment services. for example, in the Vercel one, the `outputDirectory ` is mentioned in the vercel config file. And also, when using older SDKs, Expo webpack generates `web-build` as a directory and not `dist`.

So removing this here. Feel free to create a new PR, if you think updates made to amplify are applicable to other deployment services

<Step label="1">

> Add the [**amplify-explicit.yml**](https://github.com/expo/amplify-demo/blob/master/amplify-explicit.yml) file to the root of your repo. Ensure you have removed the generated dist directory from the .gitignore file and commit the changes.

Reviewer: ```suggestion
Add the [**amplify-explicit.yml**](https://github.com/expo/amplify-demo/blob/master/amplify-explicit.yml) file to the root of your repository. Ensure you have removed the generated **dist** directory from the **.gitignore** file and committed those changes.
```

<Step label="3">

Login to the [Amplify Console](https://console.aws.amazon.com/amplify/home) and select an existing app or create a new app. Grant Amplify permission to read from your GitHub account or the organization that owns your repo.

</Step>

<Step label="4">

> Add your repo, select the branch and select "Connecting a monorepo?" to enter the path to the dist directory of your app. Choose **Next**. The Amplify Console will detect that the **amplify.yml** file is in your repo. Select "Allow AWS Amplify to automatically deploy all files hosted in your project root directory". Choose **Next**.

Reviewer: ```suggestion
Add your repo, select the branch, and select **Connecting a monorepo?** to enter the path to your app's **dist** directory. Choose **Next**. 

The Amplify Console will detect the **amplify.yml** file in your project. Select **Allow AWS Amplify to automatically deploy all files hosted in your project root directory** and choose **Next**.
```

</Step>

<Step label="5">

> Review your settings and choose **Save and deploy**. Your app will now be deployed to a `https://branchname.xxxxxx.amplifyapp.com` URL. You can now visit your web app, deploy another branch, and/or add a unified backend environment for use across your Expo Mobile and Web Apps. Follow the steps in the "Learn how to get the most out of Amplify Hosting" drop-down to "Add a custom domain with a free SSL certificate" and more.

Reviewer: ```suggestion
Review your settings and choose **Save and deploy**. Your app will now be deployed to a `https://branchname.xxxxxx.amplifyapp.com` URL. You can now visit your web app, deploy another branch, or add a unified backend environment across your Expo mobile and web apps. 

Follow the steps in the **Learn how to get the most out of Amplify Hosting** drop-down to **Add a custom domain with a free SSL certificate** and more information.
```
